lib/src/command/barback.dart

```diff
@@ -2,21 +2,31 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import '../command.dart';
+import 'package:args/command_runner.dart';
+
 import '../log.dart' as log;
 import '../utils.dart';
 
 /// Shared base class for [BuildCommand] and [ServeCommand].
-abstract class BarbackCommand extends PubCommand {
+abstract class BarbackCommand extends Command {
```

Comment:
I'm worried that making this not extend `PubCommand` could cause problems later on if we need to rely on all commands implementing that interface. Why can't you just add hidden options that match the old options?

Reply:
Mostly because there were a lot of options so this was _just_ allowing anything and the alternative is a lot more code.

It also means that if you misspell and option you'll get a misleading usage exception making you think if you correct it it will work only to find a new failure when you fix the options.

I can add them back though.

Reply:
In case it helps I _could_ make this extend `PubCommand` again...

Reply:
Can you override `PubCommand.argParser` to return an `ArgParser.allowAnything()`?

Reply:
no, `ArgParser.allowAnything()` doesn't let you set options on it and since it's a `Command` it automatically gets a `help`.

`Unsupported operation: ArgParser.allowAnything().addFlag() isn't supported.`